pkg/bloomgateway/client.go

```diff
@@ -269,8 +257,19 @@ func (c *GatewayClient) FilterChunks(ctx context.Context, _ string, interval blo
 			}
 			resp, err := client.FilterChunkRefs(ctx, req)
 			if err != nil {
-				return err
+				// We don't want a single bloom-gw failure to fail the entire query,
+				// so instrument & move on
+				c.metrics.clientRequests.WithLabelValues(typeError).Inc()
+				level.Error(c.logger).Log(
+					"msg", "filter failed for instance, skipping",
+					"addr", rs.addr,
+					"series", len(rs.groups),
+					"blocks", len(rs.blocks),
+					"err", err,
+				)
+				return nil
```

Comment:
In case a request fails, the results entries need to be equal to the request entries, otherwise we assume that all requested chunks have been filtered out.

```suggestion
				results[i] = rs.groups
				count += len(rs.groups)
				return nil
```

Reply:
Increasing `count` is not strictly needed, but the value is used for the allocation of the result buffer.